common/ECChannel.cpp

```diff
@@ -44,7 +46,8 @@ Creating a self-signed test certificate:
 openssl req -new -x509 -key privkey.pem -out cacert.pem -days 1095
 */
 
-std::atomic<SSL_CTX *> ECChannel::lpCTX{nullptr};
+std::shared_mutex ECChannel::ctx_lock;
```

Comment:
The shared_mutex is _not_ for the sake of HrSetCtx, but for the sake of HrEnableTLS, which would otherwise block any new incoming SSL instance creation (see the later patch) while another thread (connection) is busy creating its SSL instance. HrSetCtx always takes only unique_locks anyway, so the lock type is irrelevant for HrSetCtx.

You'd have to time how long SSL_new(ctx) actually takes to run when changing this to a unique_mutex, and whether that overhead is relevant (as all SSL_new(ctx) would have to be serialized), and I guessed that it's non-negligible.